src/main/java/net/sargue/time/jsptags/FormatSupport.java

```diff
@@ -13,149 +13,160 @@
-    public void setVar(String var) {
-        this.var = var;
-    }
-
-    @SuppressWarnings("UnusedDeclaration")
-    public void setScope(String scope) {
-        this.scope = Util.getScope(scope);
-    }
-
-    /*
-     * Formats the given instant or partial.
-     */
-    public int doEndTag() throws JspException {
-        if (value == null) {
-            if (var != null) {
-                pageContext.removeAttribute(var, scope);
-            }
-            return EVAL_PAGE;
-        }
-
-        // Create formatter
-        DateTimeFormatter formatter;
-        if (pattern != null) {
-            formatter = DateTimeFormatter.ofPattern(pattern);
-        } else if (style != null) {
-            formatter = Util.createFormatterForStyle(style);
-        } else {
-            // use a medium date (no time) style by default; same as jstl
-            formatter = Util.createFormatterForStyle("M-");
-        }
-
-        // set formatter locale
-        Locale locale = this.locale;
-        if (locale == null) {
-            locale = Util.getFormattingLocale(pageContext, true,
-                    DateFormat.getAvailableLocales());
-        }
-        if (locale != null) {
-            formatter = formatter.withLocale(locale);
-        }
-
-        // set formatter timezone
-        ZoneId zoneId = this.zoneId;
-        if (zoneId == null) {
-            zoneId = ZoneIdSupport.getZoneId(pageContext, this);
-        }
-        if (zoneId != null) {
-            formatter = formatter.withZone(zoneId);
-        } else {
-            if (value instanceof Instant ||
-                value instanceof LocalDateTime ||
-                value instanceof OffsetDateTime ||
-                value instanceof OffsetTime ||
-                value instanceof LocalTime)
-                // these time objects may need a zone to resolve some patterns
-                // and/or styles, and as there is no zone we revert to the
-                // system default zone
-                formatter = formatter.withZone(ZoneId.systemDefault());
-        }
-
-        // format value
-        String formatted;
-        if (value instanceof TemporalAccessor) {
-            formatted = formatter.format((TemporalAccessor) value);
-        } else {
-            throw new JspException(
-                "value attribute of format tag must be a TemporalAccessor," +
-                " was: " + value.getClass().getName()); 
-        }
-
-        if (var != null) {
-            pageContext.setAttribute(var, formatted, scope);
-        } else {
-            try {
-                pageContext.getOut().print(formatted);
-            } catch (IOException ioe) {
-                throw new JspTagException(ioe.toString(), ioe);
-            }
-        }
-
-        return EVAL_PAGE;
-    }
-
-    // Releases any resources we may have (or inherit)
-    public void release() {
-        init();
-    }
+	private static final long serialVersionUID = 1L;
```

Comment:
This class is not serializable, no need for this field.

Reply:
Actually it is because it extends TagSupport. https://tomcat.apache.org/tomcat-10.1-doc/jspapi/jakarta/servlet/jsp/tagext/TagSupport.html

Reply:
Indeed.